test/deploy_changes/common/common_helpers.go

```diff
@@ -0,0 +1,57 @@
+package common
+
+import (
+	"bytes"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/loft-sh/vcluster/test/framework"
+	"github.com/onsi/gomega"
+)
+
+const (
+	pollingInterval     = time.Second * 2
+	pollingDurationLong = time.Minute * 2
+	filePath            = "commonValues.yaml"
+	chartPath           = "../../../chart"
+)
+
+func ReplaceYAMLPlaceholders() {
+	replaceCmd := exec.Command("sed", "-i", "s|REPLACE_REPOSITORY_NAME|"+os.Getenv("REPOSITORY_NAME")+"|g", filePath)
+	err := replaceCmd.Run()
+	framework.ExpectNoError(err, "Failure to edit file")
+
+	replaceCmd = exec.Command("sed", "-i", "s|REPLACE_TAG_NAME|"+os.Getenv("TAG_NAME")+"|g", filePath)
+	err = replaceCmd.Run()
+	framework.ExpectNoError(err, "Failure to edit file")
+}
+
+func DeployChangesToVClusterUsingCLI(f *framework.Framework) {
+	gomega.Eventually(func() bool {
+		stdout := &bytes.Buffer{}
+		deployCmd := exec.Command("vcluster", "create", "--upgrade", f.VclusterName, "--namespace", f.VclusterNamespace, "--local-chart-dir", chartPath, "-f", filePath)
+		deployCmd.Stdout = stdout
+		err := deployCmd.Run()
+		framework.ExpectNoError(err)
+		return err == nil && strings.Contains(stdout.String(), "Switched active kube context to")
+	}).WithPolling(pollingInterval).WithTimeout(pollingDurationLong).Should(gomega.BeTrue())
+}
+
+func DisconnectFromVCluster() {
+	disconnectCmd := exec.Command("vcluster", "disconnect")
+	err := disconnectCmd.Run()
+	if err != nil && !strings.Contains(err.Error(), "not a virtual cluster context") {
+		framework.ExpectNoError(err, "Error disconnecting from vCluster")
+	}
+}
+
+func VerifyClusterIsActive(f *framework.Framework) {
+	gomega.Eventually(func() bool {
+		checkCmd := exec.Command("vcluster", "list")
+		output, err := checkCmd.CombinedOutput()
+		framework.ExpectNoError(err)
+		return err == nil && strings.Contains(string(output), f.VclusterName) && strings.Contains(string(output), "Running")
```

Comment:
I think this check is too general. Here is the output of my `vcluster list`:
```
      NAME    |        NAMESPACE         | STATUS  |    VERSION     | CONNECTED |    AGE     
  ------------+--------------------------+---------+----------------+-----------+------------
    vcluster2 | loft-default-v-vcluster2 | Running | 0.20.0-beta.15 |           | 266h4m40s  
    vcluster  | vcluster                 | Running | uLmOpUH        |           | 238h42m5s  
```

Let's say instead `vcluster2` was not in `Running` and f.VclusterName was `vcluster2`. The output still contains `vcluster2` and `Running`. I think instead you may want to split by new line or, even better use a regex. I still would be worried about using contains with the former because you have to consider on a single line that f.VclusterName could be a substring of another vcluster name, like with `vcluster` and `vcluster2`.